packages/payments-plugin/src/paypal/paypal-authorization.service.ts

```diff
@@ -0,0 +1,149 @@
+import { Inject } from '@nestjs/common';
+import { Logger, PaymentMethodService, RequestContext } from '@vendure/core';
+
+import { loggerCtx, PAYPAL_PAYMENT_PLUGIN_OPTIONS } from './constants';
+import { PayPalBaseService } from './paypal-base.service';
+import { PayPalError } from './paypal.error';
+import { PayPalOrderInformation, PayPalPluginOptions } from './types';
+
+export class PayPalAuthorizationService extends PayPalBaseService {
+    constructor(
+        @Inject(PAYPAL_PAYMENT_PLUGIN_OPTIONS) options: PayPalPluginOptions,
+        paymentMethodService: PaymentMethodService,
+    ) {
+        super(paymentMethodService, options);
+    }
+
+    async authorizationDetails(
+        ctx: RequestContext,
+        authorizationId: string,
+    ): Promise<PayPalOrderInformation> {
+        const accessToken = (await this.authenticate(ctx)).access_token;
+
+        try {
+            const response = await fetch(`${this.options.apiUrl}/v2/checkout/orders/${authorizationId}`, {
+                headers: {
+                    Authorization: `Bearer ${accessToken}`,
+                },
+            });
+
+            if (!response.ok) {
+                throw response;
+            }
+
+            return (await response.json()) as PayPalOrderInformation;
+        } catch (error: any) {
+            const message = 'error.paypal-authorization-details-failed';
```

Comment:
Even though SonarQube already mentioned this code is often duplicated, I want to make sure that errors are correctly handled this way in Vendure. Maybe you guys have some feedback on this :)